File: ServerSpace/src/main/java/DataLayer/PersonaNaturalDB.java
package DataLayer;

import EntityLayer.PersonaNaturalEntity;
import Enumerados.ProcessActionEnum;
import Framework.BaseDB;
import Framework.Inj;
import Framework.Utilidades;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class PersonaNaturalDB extends BaseDB {

    public ArrayList<PersonaNaturalEntity> ObtenerItem(int PersonaNaturalId) {

        ArrayList<PersonaNaturalEntity> DatoMemoria = new ArrayList<>();
        PersonaNaturalEntity en;
        try {
            Inj.Sp("sp_PersonaObtenerItem");
            Inj.Pmt_Integer("@PersonaNaturalId", PersonaNaturalId, false);
            ResultSet rs = Inj.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new PersonaNaturalEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public PersonaNaturalEntity Registrar(PersonaNaturalEntity entity) {
        Boolean State = null;
        try {
            Inj.IniciarTranssaccion(false);
            String Store = "sp_PersonaRegistar";
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Store = "sp_PersonaActualizar";
            }
            Inj.Sp(Store);
            Inj.Pmt_Integer("@PersonaNaturalId", entity.getPersonaNaturalId(), true);
            Inj.Pmt_Integer("@TipoDocumentoIdentidadId", entity.getTipoDocumentoIdentidadId(), false);
            Inj.Pmt_String("@NumDocumento", entity.getNumDocumento(), false);
            Inj.Pmt_String("@Nombres", entity.getNombres(), false);
            Inj.Pmt_String("@ApellidoPaterno", entity.getApellidoPaterno(), false);
            Inj.Pmt_String("@ApellidoMaterno", entity.getApellidoMaterno(), false);
            Inj.Pmt_String("@FechaNacimiento", entity.getFechaNacimiento().toString(), false);
            Inj.Pmt_String("@Direccion", entity.getDireccion(), false);
            Inj.Pmt_String("@Telefono", entity.getTelefono(), false);
            Inj.Pmt_String("@Correo", entity.getCorreo(), false);
            Inj.Pmt_Integer("@SexoId", entity.getSexoId(), false);
            Inj.Pmt_Integer("@EstadoCivilId", entity.getEstadoCivilId(), false);
            Inj.Pmt_Integer("@UbigeoId", entity.getUbigeoId(), false);
            Inj.Pmt_String("@CodUsuario", entity.getCodUsuario(), false);
            Inj.Pmt_String("@FechaRegistro", Utilidades.getFechaRegistro(), false);
            if (entity.getAction() == ProcessActionEnum.Add.getValor()) {
                int Id = Inj.RunInsert();
                State = Id > 0;
                if (State) {
                    entity.setPersonaNaturalId(Id);
                }
            }
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Inj.RunUpdate();
            }

        } catch (Exception ex) {
            throw new UnsupportedOperationException("Datalater : " + ex);
        }
        return entity;
    }

    public ArrayList<PersonaNaturalEntity> ObtenerMain() {

        ArrayList<PersonaNaturalEntity> DatoMemoria = new ArrayList<>();
        PersonaNaturalEntity en;
        try {
            Inj.Sp("sp_PersonaObtenerMain");
            ResultSet rs = Inj.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new PersonaNaturalEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }


}

File: Space/Space.DataLayer/PersonaNaturalDB.cs
using Framework;
using Space.EntityLayer;
using System.Data;
using System.Data.SqlClient;

namespace Space.DataLayer
{
    public class PersonaNaturalDB : BaseDB
    {

        public List<PersonaNaturalEntity> ObtenerItem(Int32 PersonaNaturalId)
        {
            try
            {
                List<PersonaNaturalEntity> ListaData = new List<PersonaNaturalEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_PersonaObtenerItem");
                Conexion.AddParemeter("PersonaNaturaId", SqlDbType.Int, 10, ParameterDirection.Input, PersonaNaturalId);
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    PersonaNaturalEntity entity = new PersonaNaturalEntity();
                    if (FillFrom<PersonaNaturalEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                Conexion.CloseConnection();
                return ListaData;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public Int32 Registrar(PersonaNaturalEntity Ent)
        {
            try
            {
                String NameStore = "sp_PersonaRegistar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("PersonaNaturalId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.PersonaNaturalId);
                Conexion.AddParemeter("TipoDocumentoIdentidadId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoDocumentoIdentidadId);
                Conexion.AddParemeter("NumDocumento", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.NumDocumento);
                Conexion.AddParemeter("Nombres", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.Nombres);
                Conexion.AddParemeter("ApellidoPaterno", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.ApellidoPaterno);
                Conexion.AddParemeter("ApellidoMaterno", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.ApellidoMaterno);
                Conexion.AddParemeter("FechaNacimiento", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaNacimiento);
                Conexion.AddParemeter("Direccion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Direccion);
                Conexion.AddParemeter("Telefono", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Telefono);
                Conexion.AddParemeter("Correo", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.Correo);
                Conexion.AddParemeter("SexoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.SexoId);
                Conexion.AddParemeter("EstadoCivilId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.EstadoCivilId);
                Conexion.AddParemeter("UbigeoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.UbigeoId);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Ent.PersonaNaturalId = Conexion.ExecuteQueryItem("PersonaNaturalId");

                return Ent.PersonaNaturalId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }


        public Int32 Actualizar(PersonaNaturalEntity Ent)
        {
            try
            {

                String NameStore = "sp_PersonaActualizar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("PersonaNaturalId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.PersonaNaturalId);
                Conexion.AddParemeter("TipoDocumentoIdentidadId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoDocumentoIdentidadId);
                Conexion.AddParemeter("NumDocumento", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.NumDocumento);
                Conexion.AddParemeter("Nombres", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.Nombres);
                Conexion.AddParemeter("ApellidoPaterno", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.ApellidoPaterno);
                Conexion.AddParemeter("ApellidoMaterno", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.ApellidoMaterno);
                Conexion.AddParemeter("FechaNacimiento", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaNacimiento);
                Conexion.AddParemeter("Direccion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Direccion);
                Conexion.AddParemeter("Telefono", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.Telefono);
                Conexion.AddParemeter("Correo", SqlDbType.VarChar, 50, ParameterDirection.Input, Ent.Correo);
                Conexion.AddParemeter("SexoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.SexoId);
                Conexion.AddParemeter("EstadoCivilId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.EstadoCivilId);
                Conexion.AddParemeter("UbigeoId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.UbigeoId);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Ent.PersonaNaturalId = Conexion.ExecuteQueryItem("PersonaNaturalId");

                return Ent.PersonaNaturalId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }

        public List<PersonaNaturalEntity> ObtenerMain()
        {
            try
            {
                List<PersonaNaturalEntity> ListaData = new List<PersonaNaturalEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_PersonaObtenerMain");
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    PersonaNaturalEntity entity = new PersonaNaturalEntity();
                    if (FillFrom<PersonaNaturalEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                Conexion.CloseConnection();
                return ListaData;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }
    }
}
